dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -77,11 +87,31 @@ class AppUpgrade extends React.Component<IAppUpgradeProps, IAppUpgradeState> {
             resource="App Repositories"
           />
         );
+      } else if (repos.length === 0) {
+        return (
+          <ErrorSelector
+            error={new Error("Unable to any repo to upgrade from")}
+            resource={releaseName}
+          />
+        );
       } else {
-        return <LoadingWrapper />;
```

Comment:
This call would show the loading wrapper forever since we have already fetched all the elements but we still don't have the info needed so I have added an error case (respos === []) and an unexpected error by default.

Reply:
Is the last case (unexpected error by default) at all possible? It seems to me like we'd never reach this code? If we will, do we have a test case for it?

Reply:
Hmm, I almost wonder if it would be better to just render an Unexpected error in that case? I guess there would be an issue with Tiller or a bug in Kubeapps if we get to this point, and I'm not sure that the proposed error message "Unable to obtain the required information to upgrade" is going to be very useful to the user. The Unexpected error directs users to file an issue in GitHub if they think they encountered a bug.

Reply:
Yes, the default case for `ErrorSelector` is to use the `UnexpectedError` component. Here, since the error type is generic, it shows the generic message but we include a little bit more of info:

![Screenshot from 2019-03-13 10-55-17](https://user-images.githubusercontent.com/4025665/54270126-f7f90a00-457e-11e9-8cde-2ab7e4191d86.png)

Reply:
Ah I see, thanks for the explanation!